R/abundance_colocalization_plot.R

```diff
@@ -0,0 +1,183 @@
+#' @param coord_fixed Fix the aspect ratio of the plot.
+#' @param pt_size Size or size range of the points.
+#' @param draw_origo Draw lines through the origin (0, 0) in the plot.
+#' @param layer Name of layer to plot. If NULL, the default layer is used.
+#' @param coloc_score Name of the colocalization score to plot.
+#' @param colors Colors to use for the colocalization score.
+#'
+#' @return A ggplot object.
+#'
+#' @examples
+#'
+#' library(pixelatorR)
+#'
+#' # Load example data as a Seurat object
+#' pxl_file <- system.file("extdata/five_cells",
+#'                         "five_cells.pxl",
+#'                         package = "pixelatorR")
+#' seur_obj <- ReadMPX_Seurat(pxl_file)
+#'
+#' # Plot with data.frame
+#' AbundanceColocalizationPlot(seur_obj, c("CD3E", "CD4"), c("CD19", "CD20"))
+#'
+#' @export
+#'
+AbundanceColocalizationPlot <- function(
+    object,
+    markers_x,
+    markers_y,
+    shared_scales = TRUE,
+    coord_fixed = TRUE,
+    pt_size = c(1, 4),
+    draw_origo= TRUE,
+    layer = NULL,
+    coloc_score = "pearson_z",
+    colors = c("#1F395F", "#496389", "#728BB1", "#AABAD1", "gray90",
+               "#E9AABF", "#CD6F8D", "#A23F5E", "#781534")
+) {
+
+  # Validate input parameters
+  stopifnot(
+    "'object' must be a Seurat object" =
+      inherits(object, "Seurat"),
+    "'markers_x' must be a character vector of length > 0" =
+      is.character(markers_x) & length(markers_x) > 0,
+    "'markers_y' must be a character vector of length > 0" =
+      is.character(markers_y) & length(markers_y) > 0,
+
+    "All markers in 'markers_x' must be present in the Seurat object" =
+      all(markers_x %in% rownames(object)),
+    "All markers in 'markers_y' must be present in the Seurat object" =
+      all(markers_y %in% rownames(object)),
+
+    "'shared_scales' must be either TRUE or FALSE" =
+      is.logical(shared_scales),
+    "'coord_fixed' must be either TRUE or FALSE" =
+      is.logical(coord_fixed),
+
+
+    "'pt_size' must be a numeric of length 1 or 2" =
+      is.numeric(pt_size) & length(pt_size) %in% c(1, 2),
+    "'draw_origo' must be either TRUE or FALSE" =
+      is.logical(draw_origo),
+    "'colors' must be a vector of colors" =
+      is.character(colors),
+
+    "'coloc_score' must be a character of length 1" =
+      is.character(coloc_score)
+  )
+
+
+  # Get data
+  plot_data <-
+    FetchData(object, vars = c(markers_x, markers_y),
+              layer = layer) %>%
+    as_tibble(rownames = "component")
+
+  plot_data <-
+    plot_data %>%
+    pivot_longer(cols = all_of(markers_x),
+                 names_to = "marker_x",
+                 values_to = "value_x") %>%
+    pivot_longer(cols = all_of(markers_y),
+                 names_to = "marker_y",
+                 values_to = "value_y")
+
+
+
+  coloc_scores <-
+    ColocalizationScores(object)
+
+  if(!coloc_score %in% colnames(coloc_scores)) {
+    stop(glue::glue("'{coloc_score}' is not a valid colocalization score."))
```

Comment:
```suggestion
    abort(glue("'{coloc_score}' is not a valid colocalization score."))
```

Reply:
We have been using `rlang` to handle errors and warnings, so I think we should do that here as well!